### PR TITLE
Fix Lambda layers test and bump Swiftshader

### PR DESCRIPTION
**Pull request by rsuderman**
Changes to swiftshader head caused an accuracy regression. This is
because x**2 is numerically computationally complex. Changed the
computation to be x*x which is computationally identical while not
depending on an exponential operator.

Bumped swiftshader to this version.

### CODE REVIEW

File: build_tools/third_party/swiftshader/build_vk_swiftshader.sh [hunk 1]
```diff
@@ -54,7 +54,7 @@ else
   SWIFTSHADER_INSTALL_DIR="${BASE_DIR?}"'\.swiftshader'
 fi
 
-SWIFTSHADER_COMMIT=c27e99245d423648fe5f0eafa73d49ce7b53d201
+SWIFTSHADER_COMMIT=57eca9df5dc56d290f662ace7481ed758618d559
```
Comment (ScottTodd) [inline build_tools/third_party/swiftshader/build_vk_swiftshader.sh:57]: This should be (ideally) kept in sync with https://github.com/google/iree/blob/47e45d3c433a4b20eb9ef28c2340e47ae7108ca0/build_tools/docker/swiftshader/Dockerfile#L12 which requires rebuilding the Docker images.

See https://github.com/google/iree/commit/dd1d96c8feffd693d64a73436d60d249570b8be5 for a full commit bumping SwiftShader

File: integrations/tensorflow/test/python/iree_tf_tests/layers/layers_test.py [hunk 2]
```diff
@@ -298,7 +298,7 @@ def has_mutually_exclusive_kwargs(
     "Lambda":
         tf_test_utils.unit_test_specs_from_signatures(
             signature_shapes=UNARY_SIGNATURE_SHAPES,
-            kwargs_to_values=dict(function=[lambda x: x**2])),
+            kwargs_to_values=dict(function=[lambda x: x*x])),
```
Comment (GMNGeoffrey) [inline integrations/tensorflow/test/python/iree_tf_tests/layers/layers_test.py]: AFAICT the exact computation here doesn't matter at all (it's just testing lambda itself), so this seems fine

**Review by GMNGeoffrey — APPROVED** [review]

**Review by GMNGeoffrey — CHANGES_REQUESTED** [review]
Wait, I think the swiftshader commit needs to be updated in other places

File: build_tools/third_party/swiftshader/build_vk_swiftshader.sh (hunk 1 above)
Comment (GMNGeoffrey) [inline build_tools/third_party/swiftshader/build_vk_swiftshader.sh:57]: This needs to be updated in https://github.com/google/iree/blob/main/build_tools/docker/swiftshader/Dockerfile too, and all the docker images rebuilt (sorry).

See the README at https://github.com/google/iree/tree/main/build_tools/docker/

Comment (rsuderman) [reply build_tools/third_party/swiftshader/build_vk_swiftshader.sh:57]: Should be fixed.

Comment (rsuderman) [reply build_tools/third_party/swiftshader/build_vk_swiftshader.sh:57]: Images updated.

**Review by GMNGeoffrey — APPROVED** [review]
FYI, you shouldn't have needed to rebuild all the docker images. Passing `--images=swiftshader` to `manage_images.py` should have been sufficient. Not a big deal if everything passes though

**Pull request merged**